packages/block-library/src/navigation/edit/unsaved-inner-blocks.js

```diff
@@ -46,41 +97,83 @@ export default function UnsavedInnerBlocks( {
 		[ blocks, serialize, saveEntityRecord ]
 	);
 
+	// Automatically save the uncontrolled blocks.
+	useEffect( async () => {
+		// The block will be disabled when used in a BlockPreview.
+		// In this case avoid automatic creation of a wp_navigation post.
+		// Otherwise the user will be spammed with lots of menus!
+		//
+		// Also ensure other navigation menus have loaded so an
+		// accurate name can be created.
+		//
+		// Don't try saving when another save is already
+		// in progress.
+		//
+		// And finally only create the menu when the block is selected,
+		// which is an indication they want to start editing.
+		if (
+			hasSavedUnsavedInnerBlocks ||
+			isDisabled ||
+			isSaving ||
+			savingLock.current ||
+			! hasResolvedDraftNavigationMenus ||
+			! hasResolvedNavigationMenus ||
+			! hasSelection
+		) {
```

Comment:
I find the amount that's going on here a bit bonkers, but I don't have another solution. There's some extreme care needed to ensure the menus are only created once.